src/org/labkey/test/tests/SampleSetParentColumn.java

```diff
@@ -0,0 +1,619 @@
+        checkAllRowsInDataRegion("Runs", "Name", List.of("Derive 2 samples from SIB_0"));
+
+    }
+
+    @Test
+    public void testMultipleParentColumns()
+    {
+
+        final String PARENT_COLUMN_SUB = "Parent-Sub-Folder";
+        final String PARENT_COLUMN_CONTAINER = "Parent Parent Folder";
+        final String SAMPLE_SET_NAME = "SimpleSampleSet05";
+
+        // Add the Alias column as a regression test.
+        String sampleText = "Name\tAlias\t" + PARENT_COLUMN_SUB + "\t" + PARENT_COLUMN_CONTAINER + "\n" +
+                "SE_01\t\n" +
+                "SE_02\t\t\tS_10\n" +
+                "SE_03\t\tSE_01\n" +
+                "SE_04\n" +
+                "SE_05\t\tSE_04\tS_11\n";
+
+        goToProjectHome();
+        projectMenu().navigateToFolder(PROJECT_NAME, SUB_FOLDER_NAME);
+
+        createSampleSetAndSetParentColumn(SAMPLE_SET_NAME,
+                PROJECT_NAME + "/" + SUB_FOLDER_NAME,
+                Map.of(PARENT_COLUMN_CONTAINER, PARENT_CONTAINER_SAMPLE_SET_NAME,
+                        PARENT_COLUMN_SUB, SAMPLE_SET_NAME));
+
+        log("Import samples that have an alias column.");
+        SampleSetHelper sampleHelper = new SampleSetHelper(this);
+        sampleHelper.bulkImport(sampleText);
+
+        log("Check sample 'SE_02' and make sure the parent materials are correct. It's parent should be in the parent container.");
+        waitAndClickAndWait(Locator.linkWithText("SE_02"));
+        checkAllRowsInDataRegion("parentMaterials", "Name", List.of("S_10"));
+        checkAllRowsInDataRegion("parentMaterials", "Run", List.of(" "));
+        checkAllRowsInDataRegion("Runs", "Name", List.of("Derive sample from S_10"));
+
+        clickAndWait(Locator.linkWithText(SAMPLE_SET_NAME));
+
+        log("Check sample 'SE_03' and make sure the parent materials are correct. It's parent should be in the same sample set.");
+        waitAndClickAndWait(Locator.linkWithText("SE_03"));
+        checkAllRowsInDataRegion("parentMaterials", "Name", List.of("SE_01"));
+        checkAllRowsInDataRegion("parentMaterials", "Run", List.of(" "));
+        checkAllRowsInDataRegion("Runs", "Name", List.of("Derive sample from SE_01"));
+
+        clickAndWait(Locator.linkWithText(SAMPLE_SET_NAME));
+
+        log("Check sample 'SE_05' which should have two parents, one in this sample set and another in the parent container sample set .");
+        waitAndClickAndWait(Locator.linkWithText("SE_05"));
+        checkAllRowsInDataRegion("parentMaterials", "Name", List.of("SE_04", "S_11"));
+        checkAllRowsInDataRegion("parentMaterials", "Run", List.of(" ", " "));
+        checkAllRowsInDataRegion("parentMaterials", "Sample Set", List.of(SAMPLE_SET_NAME, PARENT_CONTAINER_SAMPLE_SET_NAME));
+
+        // Not sure how reliable this test will be, will the order always be "S_11, SE_04"?
+        checkAllRowsInDataRegion("Runs", "Name", List.of("Derive sample from S_11, SE_04"));
+
+    }
+
+    @Test
```

Comment:
/**
Check that the alias column does not conflict with parent mappings (Issue 37946)
**/

Reply:
Added this comment.